client/state/media/reducer.js

```diff
@@ -36,6 +40,52 @@ export const items = createReducer( {}, {
 	}
 } );
 
+export const queries = ( () => {
+	function applyToManager( state, siteId, method, createDefault, ...args ) {
+		if ( ! state[ siteId ] ) {
+			if ( ! createDefault ) {
+				return state;
+			}
+
+			return {
+				...state,
+				[ siteId ]: ( new MediaQueryManager() )[ method ]( ...args )
+			};
+		}
+
+		const nextManager = state[ siteId ][ method ]( ...args );
+
+		if ( nextManager === state[ siteId ] ) {
+			return state;
+		}
+
+		return {
+			...state,
+			[ siteId ]: nextManager
+		};
+	}
+
+	return createReducer( {}, {
+		[ MEDIA_RECEIVE ]: ( state, { siteId, media, found, query } ) => {
+			return applyToManager( state, siteId, 'receive', true, media, { found, query } );
+		}
+	} );
+} )();
+
+export const queryRequests = keyedReducer( 'siteId', createReducer( [], {
```

Comment:
It's really not obvious, and I encountered it myself the hard way, but composing a `keyedReducer` `createReducer` will persist its state even though one would expect `createReducer` to only persist if a schema is provided.

Related: https://github.com/Automattic/wp-calypso/pull/10016#issuecomment-269004720